gridfs/asynchronous/grid_file.py

```diff
@@ -1486,13 +1489,14 @@ def __init__(
     _file: Any
     _chunk_iter: Any
 
-    async def __anext__(self) -> bytes:
-        return super().__next__()
+    if not _IS_SYNC:
 
-    def __next__(self) -> bytes:  # noqa: F811, RUF100
-        if _IS_SYNC:
-            return super().__next__()
-        else:
+        async def __anext__(self) -> bytes:
+            return await self.readline()
```

Comment:
We need to handle the end-of-stream and raise StopAsyncIteration here. Could you add a test for this?